third_party/move/move-compiler/transactional-tests/tests/control_flow/for_type_mismatch.exp

```diff
@@ -0,0 +1,40 @@
+processed 1 task
+
+task 0 'run'. lines 1-10:
+Error: error[E04007]: incompatible types
+  ┌─ TEMPFILE:5:9
+  │    
+5 │ ╭ ╭         for (i in true..false) {
```

Comment:
This set of errors is very confusing to the user.  Can we somehow refine the set of errors we produce?

Reply:
Resolving this issue needs deeper changes. I will add it as an issue after this PR is merged.